src/java/htsjdk/variant/variantcontext/GenotypeJEXLContext.java

```diff
@@ -0,0 +1,58 @@
+package htsjdk.variant.variantcontext;
+
+import htsjdk.variant.vcf.VCFConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *
+ * @author bbimber
+ *
+ * implements the JEXL context for Genotype; this saves us from
+ * having to generate a JEXL context lookup map everytime we want to evaluate an expression.
+ *
+ */
+public class GenotypeJEXLContext extends VariantJEXLContext {
+    private Genotype g;
+
+    private interface AttributeGetter {
+        public Object get(Genotype g);
+    }
+
+    private static Map<String, AttributeGetter> attrs = new HashMap<String, AttributeGetter>();
+
+    static {
+        attrs.put("g", new AttributeGetter() { public Object get(Genotype g) { return g; }});
+        attrs.put(VCFConstants.GENOTYPE_KEY, new AttributeGetter() { public Object get(Genotype g) { return g.getGenotypeString(); }});
+
+        attrs.put("isHom", new AttributeGetter() { public Object get(Genotype g) { return g.isHom() ? "1" : "0"; }});
```

Comment:
you are correct, though the README.md states that we are on JAVA8...

On Fri, Apr 8, 2016 at 1:32 PM, bbimber <notifications@github.com> wrote:

> In src/java/htsjdk/variant/variantcontext/GenotypeJEXLContext.java
> <https://github.com/samtools/htsjdk/pull/532#discussion_r59061165>:
>
> > + *
> > + */
> > +public class GenotypeJEXLContext extends VariantJEXLContext {
> > +    private Genotype g;
> > +
> > +    private interface AttributeGetter {
> > +        public Object get(Genotype g);
> > +    }
> > +
> > +    private static Map<String, AttributeGetter> attrs = new HashMap<String, AttributeGetter>();
> > +
> > +    static {
> > +        attrs.put("g", new AttributeGetter() { public Object get(Genotype g) { return g; }});
> > +        attrs.put(VCFConstants.GENOTYPE_KEY, new AttributeGetter() { public Object get(Genotype g) { return g.getGenotypeString(); }});
> > +
> > +        attrs.put("isHom", new AttributeGetter() { public Object get(Genotype g) { return g.isHom() ? "1" : "0"; }});
>
> FWIW, the checked in project IPR file has the languageLevel as 1.6. If
> you're officially on 1.8, it might be nice to update that file (not in this
> PR).
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/samtools/htsjdk/pull/532/files/e8e473449289b9b27f478e72a75a626244eaa363#r59061165>
>